File: frontend/src/services/ItemTransactionService.js
import axios from "axios";

export default class ItemTransactionService{

    getAll() {
        return axios.get("http://localhost:8080/api/itemTransaction/getAll")
    }

    getAllSortedByStatus(){
        return axios.get("http://localhost:8080/api/itemTransaction/getAllSortedByStatus")
    }

    getById(id) {
        return axios.get(`http://localhost:8080/api/itemTransaction/getById?id=${id}`)
    }

    existsByBuyerId(buyerId){
        return axios.get(`http://localhost:8080/api/itemTransaction/existsByFkBuyerId?buyerId=${buyerId}`)
    }

    existsByEmployeeId(employeeId){
        return axios.get(`http://localhost:8080/api/itemTransaction/existsByFkEmployeeId?employeeId=${employeeId}`)
    }

    existsBySellerId(sellerId){
        return axios.get(`http://localhost:8080/api/itemTransaction/existsByFkSellerId?sellerId=${sellerId}`)
    }

    getByBuyerId(buyerId){
        return axios.get(`http://localhost:8080/api/itemTransaction/getByFkBuyerId?buyerId=${buyerId}`)
    }

    getByEmployeeId(employeeId){
        return axios.get(`http://localhost:8080/api/itemTransaction/getByFkEmployeeId?employeeId=${employeeId}`)
    }

    getBySellerId(sellerId){
        return axios.get(`http://localhost:8080/api/itemTransaction/getByFkSellerId?sellerId=${sellerId}`)
    }

    deleteById(id) {
        return axios.delete(`http://localhost:8080/api/itemTransaction/deleteById?id=${id}`)
    }

    updateStatus(values){
        return axios.patch("http://localhost:8080/api/itemTransaction/updateStatus", values)
    }
}

File: backend/src/main/java/ecrow/backend/business/abstracts/ItemTransactionService.java
package ecrow.backend.business.abstracts;

import ecrow.backend.core.utilities.results.DataResult;
import ecrow.backend.core.utilities.results.Result;
import ecrow.backend.entities.concretes.ItemTransaction;
import ecrow.backend.entities.dtos.ItemTransactionAddDto;
import ecrow.backend.entities.dtos.ItemTransactionStatusUpdateDto;

import java.util.List;

public interface ItemTransactionService {
    DataResult<List<ItemTransaction>> getAll();
    DataResult<List<ItemTransaction>> getAllSortedByStatus();
    DataResult<ItemTransaction> getById(Integer id);
    DataResult<List<ItemTransaction>> getByFkEmployeeId(Integer employeeId);
    DataResult<List<ItemTransaction>> getByFkBuyerId(Integer buyerId);
    DataResult<List<ItemTransaction>> getByFkSellerId(Integer sellerId);
    DataResult<Boolean> existsByFkEmployeeId(Integer employeeId);
    DataResult<Boolean> existsByFkBuyerId(Integer buyerId);
    DataResult<Boolean> existsByFkSellerId(Integer sellerId);
    Result deleteById(Integer id);

    Result add(ItemTransactionAddDto itemTransactionAddDto);
    Result updateStatus(ItemTransactionStatusUpdateDto itemTransactionStatusUpdateDto);
}
